plugins/org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/trace/AbstractTrace.java

```diff
@@ -526,5 +530,33 @@ protected LanguageInfo findLanguage(/* @Nullable */ URI uri) {
 		}
 		return null;
 	}
+	
+	@Override
+	public String toString() {
+		return new AbstractTraceRegionToString() {
+
+			@Override
+			protected AbstractTraceRegion getTrace() {
+				return rootTraceRegion;
+			}
+
+			@Override
+			public String getText(URI uri) {
+				try {
+					return new String(ByteStreams.toByteArray(getContents(uri, getLocalProject())));
```

Comment:
This appears to be a lurking encoding issue. Since we do a lot of offset voodoo, this could have strange impact on the result